index.bs

```diff
@@ -2447,32 +4021,64 @@ partial interface MLGraphBuilder {
-            matrices with dimensions corresponding to the last two indices. The
-            matrix multiplication will be broadcasted accordingly by following
-            [[!numpy-broadcasting-rule]]. The output is a N-D tensor whose rank
-            is the maximum rank of the input tensors. For each dimension, except
-            the last two, of the output tensor, its size is the maximum size
-            along that dimension of the input tensors.
-        - If *a* is 1-D, it is converted to a 2-D tensor by prepending a 1 to
-            its dimensions.
-        - If *b* is 1-D, it is converted to a 2-D tensor by by appending a 1 to
-            its dimensions.
-        - If both *a* and *b* are 1-D, the operation is a vector dot-product,
-            which produces a scalar output.
+</div>
+<div>
+    Computes the matrix product of two input tensors as follows:
+        - If both *a* and *b* are 2-dimensional, they are multiplied like conventional
+            matrices and produce a 2-dimensional tensor as the output.
+        - If either *a* or *b* is `N`-dimensional where `N > 2`, it is treated as a stack of matrices with dimensions corresponding to the last two indices. The matrix multiplication will be broadcasted accordingly by following the [[!numpy-broadcasting-rule]]. The output is a `N`-dimensional tensor whose rank is the maximum [=rank=] of the input tensors. For each dimension, except the last two, of the output tensor, its size is the maximum size along that dimension of the input tensors.
+        - If *a* is 1-dimensional, it is converted to a 2-dimensional tensor by prepending a 1 to its dimensions.
+        - If *b* is 1-dimensional, it is converted to a 2-dimensional tensor by by appending a 1 to its dimensions.
+        - If both *a* and *b* are 1-dimensional, the operation is a vector dot-product, which produces a scalar output.
 </div>
 
+<details open>
+  <summary>
+    To <dfn dfn-for=MLGraphBuilder>calculate matmul output sizes</dfn>, given |a| and |b| run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. Let |shapeA| be |a|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} and |sizeA| the [=list/size=] of |shapeA|.
+    1. Let |shapeB| be |a|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} and |sizeB| the [=list/size=] of |shapeB|.
+    1. If |sizeA| and |sizeB| is `1`, return `«  1  »`.
+    1. If | sizeA| is `1` and |sizeB| is not, then insert `1` in the front of |shapeA| to become [ 1 | |shapeA| ] and let |sizeA| be `2`.
+    1. If | sizeB| is `1` and |sizeA| is not, then insert `1` in the front of |shapeB| to become [ 1 | |shapeB| ] and let |sizeB| be `2`.
+    1. Let |shape| be an array whose size |size| is the maximum of |sizeA| and |sizeB|.
+    1. [=map/For each=] |index| between 0 and |size|:
+       1. Set |shape|[|index|] to the maximum of |shapeA|[|index|] and |shapeB|[|index|].
```

Comment:
Needs one more space to show up as a nested list.

Reply:
Sorry I don't get this one. Space between `[` .. `]` and `|`? Renders the same for me.

Reply:
The whole line needs to be indented by one more space for Bikeshed to render it as a sub-list.

Right now it's rendered as a single-level list in the HTML.